### PR TITLE
PR #527: Module organisation structure changed.

### PR DESCRIPTION
**Pull request by vadeg**
## Issue link

#527 module structure

## Purpose
- `proto` and `spring` integration moved to `dozer-integrations`
- `proto` renamed to `proto3`
- `spring` renamed to `dozer-spring4`
- Tests fixed.

### CODE REVIEW

**Comment by vadeg** [comment]
@garethahealy @orange-buffalo could you please review?

File: dozer-integrations/dozer-proto3/pom.xml [hunk 1]
```diff
@@ -22,15 +22,14 @@
 
     <modelVersion>4.0.0</modelVersion>
     <parent>
+        <artifactId>dozer-integrations</artifactId>
```
Comment (garethahealy) [inline dozer-integrations/dozer-proto3/pom.xml]: Can you flip the order, so groupid is above artifact

File: dozer-integrations/dozer-proto3/pom.xml [hunk 2]
```diff
@@ -22,15 +22,14 @@
...
         <groupId>com.github.dozermapper</groupId>
-        <artifactId>dozer-plugins-parent</artifactId>
         <version>6.2.0-SNAPSHOT</version>
-        <relativePath>../plugins-parent</relativePath>
```
Comment (garethahealy) [inline dozer-integrations/dozer-proto3/pom.xml]: Can you add relativePath for dozer-integrations

File: dozer-integrations/dozer-proto3/src/test/java/com/github/dozermapper/protobuf/functional_tests/ProtoBeansMappingTest.java [hunk 3]
```diff
@@ -44,10 +43,7 @@
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (garethahealy) [inline dozer-integrations/dozer-proto3/src/test/java/com/github/dozermapper/protobuf/functional_tests/ProtoBeansMappingTest.java]: Can you revert this, so everything is explicit

File: dozer-integrations/dozer-spring-support/dozer-spring-boot-autoconfigure/src/test/java/com/github/dozermaper/springboot/autoconfigure/AutoConfigurationTests.java [hunk 4]
```diff
@@ -31,10 +30,7 @@
 import org.springframework.context.ConfigurableApplicationContext;
 import org.springframework.context.annotation.Bean;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/dozer-spring-boot-autoconfigure/src/test/java/com/github/dozermaper/springboot/autoconfigure/AutoConfigurationTests.java]: Same as other comment, explicit please

File: dozer-integrations/dozer-spring-support/dozer-spring4/pom.xml [hunk 5]
```diff
@@ -23,14 +23,13 @@
     <modelVersion>4.0.0</modelVersion>
     <parent>
         <groupId>com.github.dozermapper</groupId>
-        <artifactId>dozer-plugins-parent</artifactId>
+        <artifactId>dozer-spring-support</artifactId>
         <version>6.2.0-SNAPSHOT</version>
-        <relativePath>../plugins-parent</relativePath>
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/dozer-spring4/pom.xml]: Can you add relative

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/main/resources/META-INF/spring.schemas [hunk 6]
```diff
@@ -0,0 +1 @@
+http\://dozermapper.github.io/schema/dozer-spring.xsd=../../../../../../schema/src/main/resources/schema/dozer-spring.xsd
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/dozer-spring4/src/main/resources/META-INF/spring.schemas]: is this correct? shouldn't it be path within the classpath

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/java/com/github/dozermapper/spring/DozerBeanMapperFactoryBeanTest.java [hunk 7]
```diff
@@ -1,105 +1,104 @@
-/*
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/dozer-spring4/src/test/java/com/github/dozermapper/spring/DozerBeanMapperFactoryBeanTest.java]: has anything changed? if not, can you revert to make commit cleaner.

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/java/com/github/dozermapper/spring/functional_tests/SpringIntegrationTest.java [hunk 8]
```diff
@@ -30,9 +29,7 @@
 import org.junit.Test;
 import org.springframework.context.support.ClassPathXmlApplicationContext;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/dozer-spring4/src/test/java/com/github/dozermapper/spring/functional_tests/SpringIntegrationTest.java]: explicit please

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/applicationContext.xml [hunk 9]
```diff
@@ -27,7 +27,7 @@
 
     <!-- Provided Spring Integration -->
     <bean id="byFactory" class="com.github.dozermapper.spring.DozerBeanMapperFactoryBean">
-        <property name="mappingFiles" value="classpath*:mappings/*.xml"/>
+        <property name="mappingFiles" value="classpath*:mappings/mappingSpring.xml"/>
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/applicationContext.xml]: Any reason for this change? if not, please revert

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/spring-bean-factory.xml [hunk 10]
```diff
@@ -1,39 +1,39 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/spring-bean-factory.xml]: same as other, if nothing changed, please revert

File: dozer-integrations/dozer-spring-support/pom.xml [hunk 11]
```diff
@@ -18,36 +18,32 @@
...
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-
-    <modelVersion>4.0.0</modelVersion>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <parent>
+        <artifactId>dozer-integrations</artifactId>
         <groupId>com.github.dozermapper</groupId>
-        <artifactId>dozer-plugins-parent</artifactId>
         <version>6.2.0-SNAPSHOT</version>
-        <relativePath>../plugins-parent</relativePath>
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/pom.xml]: relative please

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/springNameSpace.xml [hunk 12]
```diff
@@ -21,7 +21,7 @@
        xmlns:dozer="http://dozermapper.github.io/schema/dozer-spring"
        xsi:schemaLocation="
           http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans-4.3.xsd
-          http://dozermapper.github.io/schema/dozer-spring http://dozermapper.github.io/schema/dozer-spring.xsd">
+          http://dozermapper.github.io/schema/dozer-spring https://dozermapper.github.io/schema/dozer-spring.xsd">
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/springNameSpace.xml]: Can you add a test for this. i know HTTPS will work, but i dont want to suggest something thats not working. see:
- https://github.com/DozerMapper/dozer/pull/538#issuecomment-346020929

File: dozer-integrations/dozer-spring-support/pom.xml [hunk 13]
```diff
@@ -18,36 +18,32 @@
...
-                <artifactId>dozer-spring</artifactId>
-                <version>${project.version}</version>
-            </dependency>
         </dependencies>
     </dependencyManagement>
 
     <modules>
         <module>dozer-spring-boot-autoconfigure</module>
         <module>dozer-spring-boot-starter</module>
+        <module>dozer-spring4</module>
```
Comment (garethahealy) [inline dozer-integrations/dozer-spring-support/pom.xml]: can you move spring4 to the top please as starters use it - explicit order

File: dozer-integrations/pom.xml [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <groupId>com.github.dozermapper</groupId>
+        <artifactId>dozer-plugins-parent</artifactId>
+        <version>6.2.0-SNAPSHOT</version>
+        <relativePath>../plugins-parent</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
```
Comment (garethahealy) [inline dozer-integrations/pom.xml]: can you put model above parent pleas

File: pom.xml [hunk 15]
```diff
@@ -78,10 +78,8 @@
         <module>plugins-parent</module>
         <module>schema</module>
         <module>core</module>
-        <module>spring</module>
-        <module>dozer-spring-boot-auto-configuration</module>
-        <module>proto</module>
         <module>tests</module>
+        <module>dozer-integrations</module>
```
Comment (garethahealy) [inline pom.xml]: can you move dozer-integrations above tests please - explicit order

File: tests/dozer-osgi-tests/src/test/java/com/github/dozermapper/osgitests/DozerProtoOsgiContainerTest.java [hunk 16]
```diff
@@ -40,14 +38,8 @@
...
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
-import static org.ops4j.pax.exam.CoreOptions.composite;
-import static org.ops4j.pax.exam.CoreOptions.junitBundles;
-import static org.ops4j.pax.exam.CoreOptions.options;
-import static org.ops4j.pax.exam.CoreOptions.systemPackages;
-import static org.ops4j.pax.exam.CoreOptions.url;
+import static org.junit.Assert.*;
+import static org.ops4j.pax.exam.CoreOptions.*;
```
Comment (garethahealy) [inline tests/dozer-osgi-tests/src/test/java/com/github/dozermapper/osgitests/DozerProtoOsgiContainerTest.java]: explicit please

File: tests/dozer-osgi-tests/src/test/java/com/github/dozermapper/osgitests/DozerSpringOsgiContainerTest.java [hunk 17]
```diff
@@ -37,13 +36,9 @@
 import org.osgi.framework.BundleContext;
 
 import static com.github.dozermapper.osgitests.support.OptionsSupport.localBundle;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertNull;
+import static org.junit.Assert.*;
```
Comment (garethahealy) [inline tests/dozer-osgi-tests/src/test/java/com/github/dozermapper/osgitests/DozerSpringOsgiContainerTest.java]: explicit please

**Comment by garethahealy** [comment]
@vadeg ; can you also update the migration docs here:
- https://github.com/DozerMapper/dozer/blob/master/docs/asciidoc/migration/v61-to-v62.adoc

as per for the GAV changes:
- https://github.com/DozerMapper/dozer/blob/master/docs/asciidoc/migration/v5-to-v6.adoc

File: dozer-integrations/dozer-proto3/pom.xml (hunk 1 above)
Comment (vadeg) [reply dozer-integrations/dozer-proto3/pom.xml]: Done.

File: dozer-integrations/dozer-proto3/pom.xml (hunk 2 above)
Comment (vadeg) [reply dozer-integrations/dozer-proto3/pom.xml]: What for? `dozer-integrations` is the direct parent of current project.

File: dozer-integrations/dozer-proto3/src/test/java/com/github/dozermapper/protobuf/functional_tests/ProtoBeansMappingTest.java (hunk 3 above)
Comment (vadeg) [reply dozer-integrations/dozer-proto3/src/test/java/com/github/dozermapper/protobuf/functional_tests/ProtoBeansMappingTest.java]: yes

File: dozer-integrations/dozer-spring-support/dozer-spring-boot-autoconfigure/src/test/java/com/github/dozermaper/springboot/autoconfigure/AutoConfigurationTests.java (hunk 4 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring-boot-autoconfigure/src/test/java/com/github/dozermaper/springboot/autoconfigure/AutoConfigurationTests.java]: done

File: dozer-integrations/dozer-spring-support/dozer-spring4/pom.xml (hunk 5 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/pom.xml]: Sorry, relative to which module?

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/main/resources/META-INF/spring.schemas (hunk 6 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/src/main/resources/META-INF/spring.schemas]: No. This is not correct. Updated.

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/java/com/github/dozermapper/spring/DozerBeanMapperFactoryBeanTest.java (hunk 7 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/src/test/java/com/github/dozermapper/spring/DozerBeanMapperFactoryBeanTest.java]: Unused `import org.dozer.loader.api.BeanMappingBuilder;` was removed and together with move it seems has such effect.

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/java/com/github/dozermapper/spring/functional_tests/SpringIntegrationTest.java (hunk 8 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/src/test/java/com/github/dozermapper/spring/functional_tests/SpringIntegrationTest.java]: done

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/applicationContext.xml (hunk 9 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/applicationContext.xml]: reverted

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/springNameSpace.xml (hunk 12 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/springNameSpace.xml]: Added test.

**Comment by vadeg** [comment]
@garethahealy sorry for such a huge commit. 98% of changes related to license header update. Without this build will fail.

File: dozer-integrations/dozer-spring-support/dozer-spring4/pom.xml (hunk 5 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/pom.xml]: or you mean 
```
<relativePath/>
```
?

File: dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/spring-bean-factory.xml (hunk 10 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/src/test/resources/spring-bean-factory.xml]: Line ending was changed to `LF` automatically. Git uses `LF`.

File: dozer-integrations/dozer-spring-support/dozer-spring4/pom.xml (hunk 5 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/dozer-spring4/pom.xml]: Got it already.

File: dozer-integrations/dozer-spring-support/pom.xml (hunk 11 above)
Comment (vadeg) [reply dozer-integrations/dozer-spring-support/pom.xml]: Done

**Comment by garethahealy** [comment]
@vadeg ; can you rebase/squash. have pushed license update.

**Comment by vadeg** [comment]
@garethahealy yes. sure.

File: docs/asciidoc/migration/v5-to-v6.adoc [hunk 18]
```diff
@@ -28,6 +28,14 @@ Dozer is built against Java JDK 1.8 ONLY.
 === 3. dozer-osgi dropped
 You no longer need to use the below, if you are running in an OSGi environment:
 
+=== 4. Artifacts renamed
```
Comment (garethahealy) [inline docs/asciidoc/migration/v5-to-v6.adoc]: this migration doc should not be changed. its only for people moving from 5.x to 6.0. please revert

Comment (vadeg) [reply docs/asciidoc/migration/v5-to-v6.adoc]: Done.

**Pull request merged**